src/fund/shares/Shares.sol

```diff
@@ -248,76 +260,58 @@ contract Shares is IShares, TokenUser, Spoke, SharesToken {
-    }
+    // PRIVATE FUNCTIONS
 
     /// @notice Redeem a specified quantity of the sender's shares
-    /// for ONLY SPECIFIC ASSETS in the fund, forfeiting the remaining assets
-    /// @dev Do not call directly, unless an asset throws preventing redemption.
-    /// Calling directly with a limited set of assets will result in the sender
-    /// losing claim to the remaining assets for their shares. It is intended as
-    /// a last resort for users to directly redeem their assets.
-    /// @dev Rewards all fees prior to redemption
+    /// for a proportionate slice of the fund's assets
+    /// @dev If _bypassFailure is set to true, the user will lose their claim to any assets for
+    /// which the transfer function fails. This should always be false unless explicitly intended
     /// @param _sharesQuantity The amount of shares to redeem
-    /// @param _assets The assets to receive from the redemption
-    function redeemSharesWithConstraints(uint256 _sharesQuantity, address[] memory _assets)
-        public
-    {
-        require(_assets.length > 0, "redeemSharesWithConstraints: _assets cannot be empty");
-        require(_sharesQuantity > 0, "redeemSharesWithConstraints: _sharesQuantity must be > 0");
-
-        IFeeManager(routes.feeManager).rewardAllFees();
-
+    /// @param _bypassFailure True if token transfer failures should be ignored and forfeited
+    function __redeemShares(uint256 _sharesQuantity, bool _bypassFailure) private {
+        require(_sharesQuantity > 0, "__redeemShares: _sharesQuantity must be > 0");
         require(
             _sharesQuantity <= balanceOf(msg.sender),
-            "redeemSharesWithConstraints: _sharesQuantity exceeds sender balance"
+            "__redeemShares: _sharesQuantity exceeds sender balance"
         );
 
+        IVault vault = IVault(routes.vault);
+        address[] memory ownedAssets = vault.getOwnedAssets();
+        require(ownedAssets.length > 0, "__redeemShares: ownedAssets is empty");
+
+        IFeeManager(routes.feeManager).rewardAllFees();
+
         // Destroy the shares
         uint256 sharesSupply = totalSupply();
         _burn(msg.sender, _sharesQuantity);
 
-        // Calculate and transfer owed assets to redeemer
-        IVault vault = IVault(routes.vault);
-        uint256[] memory owedQuantities = new uint256[](_assets.length);
-        for (uint256 i = 0; i < _assets.length; i++) {
-            uint256 quantityHeld = vault.assetBalances(_assets[i]);
-            require(quantityHeld > 0, "Requested asset holdings is 0");
-
-            // Confirm asset has not already been transfered
-            for (uint256 j = 0; j < i; j++) {
-                require(
-                    _assets[i] != _assets[j],
-                    "Attempted to redeem duplicate asset"
-                );
-            }
-
+        // Calculate and transfer owned assets to redeemer
+        uint256[] memory ownedQuantities = new uint256[](ownedAssets.length);
```

Comment:
This actually isn't the "owned" quantity (the amount the fund owns), it's the "owed" quantity (the amount the user gets paid out). The namespacing is probably too similar between "owned" and "owed" though. Let's change this to `payoutAssets` and `payoutQuantities`?

Reply:
(remember to change "owned" in the comment on line 288, too)